scipy/spatial/transform/rotation.py

```diff
@@ -1639,3 +1640,308 @@ def __call__(self, times):
 
         return (self.rotations[ind] *
                 Rotation.from_rotvec(self.rotvecs[ind] * alpha[:, None]))
+
+
+def _spline_non_linear_term(rotvec, omega):
+    # R function formulation from eqn 34.
+    # Half angle formulae used to simplify calculations
+
+    # rotvec and omega should have same shape
+    dtheta = np.linalg.norm(rotvec, axis=1)
+    non_zero = (dtheta != 0)
+
+    e = np.zeros(rotvec.shape)
+    e[non_zero] = rotvec[non_zero] / dtheta[non_zero][:, None]
+
+    # All vectors are column vectors
+    small_theta = (dtheta <= 1e-6)
+    small_angle = dtheta[small_theta, None]
+    large_angle = dtheta[~small_theta, None]
+
+    # Define cot(t/2) only for large angles, define r0 and r1 for all angles
+    cot_theta_2 = 1 / np.tan(large_angle * 0.5)
+
+    r0 = np.empty((dtheta.shape[0], 1))
+    r0[~small_theta] = (large_angle / (4 * (np.sin(large_angle * 0.5) ** 2)) -
+                        0.5 * cot_theta_2)
+    r0[small_theta] = small_angle / 6 + (small_angle ** 3) / 180
+
+    r1 = np.empty((dtheta.shape[0], 1))
+    r1[~small_theta] = cot_theta_2 - 2 / large_angle
+    r1[small_theta] = -small_angle / 6 - (small_angle ** 3) / 360
+
+    # row wise dot product not performed by np.dot or np.tensordot
+    dot_prod = np.einsum('ij,ij->i', e, omega)[:, None]
+    cross_prod = np.cross(e, omega)
+
+    norms = np.linalg.norm(omega, axis=1)[:, None]
+
+    result = (r0 * (norms ** 2 - dot_prod ** 2) * e +
+              r1 * e * dot_prod * np.cross(cross_prod, e))
+    result[~non_zero] = [0, 0, 0]
+    return result
+
+
+def _spline_w_i_modifier(rotvec, omega):
+    # B function formulation from eqn 24
+    # rotvec and omega should have same shape
+    dtheta = np.linalg.norm(rotvec, axis=1)
+    non_zero = (dtheta != 0)
+
+    e = np.zeros(rotvec.shape)
+    e[non_zero] = rotvec[non_zero] / dtheta[non_zero][:, None]
+
+    dot_prod = np.einsum('ij,ij->i', omega, e)[:, None]
+    cross_prod = np.cross(e, omega)
+
+    small_theta = (dtheta <= 1e-6)
+    small_angle = dtheta[small_theta, None]
+    large_angle = dtheta[~small_theta, None]
+
+    c1 = np.empty((dtheta.shape[0], 1))
+    c1[~small_theta] = np.sin(large_angle) / large_angle
+    c1[small_theta] = 1 - (small_angle ** 2) / 6 + (small_angle ** 4) / 120
+
+    c2 = np.empty_like(c1)
+    c2[~small_theta] = (1 - np.cos(large_angle)) / large_angle
+    c2[small_theta] = small_angle / 2 - (small_angle ** 3) / 24
+
```

Comment:
The need to do this small vs. large angle calculation for `c1` and `c2` can probably be replaced by `special.sinc` (or `np.sinc`) and `special.cosm1`.

Reply:
@ewmoore nice suggestion!

Certainly can be used in a lot of places (including some previous code @adbugger wrote). In this particular case we should represent 1 - cos(x) as 2 sin^2 (x/2) and then sinc emerges.

One thing which is slightly annoying is that scipy/numpy defines sinc as sin(pi x) / (pi x), I would much prefer to have a clean definition sin x / x. I guess we may write a simple wrapper to alleviate that.

@adbugger investigate that idea. I think we can remove all small/large angle branches using sinc, here and in `Rotation` class as well.

Reply:
Will implement this in a separate PR.